Modules/TPC/src/Clusters.cxx

```diff
@@ -0,0 +1,231 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   Clusters.cxx
+/// \author Thomas Klemenz
+///
+
+#include "Framework/PartRef.h"
+#include "Framework/WorkflowSpec.h" // o2::framework::mergeInputs
+#include "Framework/DataRefUtils.h"
+#include "Framework/DataSpecUtils.h"
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "Framework/InputRecordWalker.h"
+#include "DataFormatsTPC/TPCSectorHeader.h"
+
+#include <bitset>
+
+using namespace o2::framework;
+using namespace o2::header;
+using namespace o2::tpc;
+
+// root includes
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TH2.h>
+
+// O2 includes
+#include "Framework/ProcessingContext.h"
+#include "DataFormatsTPC/Defs.h"
+//#include "TPCQC/Helpers.h"
+#include "TPCBase/Painter.h"
+#include "DataFormatsTPC/ClusterNative.h"
+#include "DataFormatsTPC/ClusterNativeHelper.h"
+#include "SimulationDataFormat/MCTruthContainer.h"
+#include "DataFormatsTPC/Constants.h"
+
+// QC includes
+#include "QualityControl/QcInfoLogger.h"
+#include "TPC/Clusters.h"
+
```

Comment:
can you try to add
`template void o2::tpc::painter::fillHistogram2D<float>(TH2& h2D, const CalDet<float>& calDet, Side side);`
to see if this solved the mac issue?

Reply:
If so I will add this in o2 and then it can be removed again, but it will be easier to quickly test it here.

Reply:
Hi @wiechula, I inserted the line in Cluster.cxx but get an error:
`Scanning dependencies of target QcTPC
[ 95%] Building CXX object Modules/TPC/CMakeFiles/QcTPC.dir/src/Clusters.cxx.o
[ 96%] Building CXX object Modules/TPC/CMakeFiles/QcTPC.dir/G__QcTPC.cxx.o
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx: In instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]':
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104:   required from here
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104: error: explicit instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]' but no definition available [-fpermissive]
 template void o2::tpc::painter::fillHistogram2D<float>(TH2& h2D, const CalDet<float>& calDet, Side side);
                                                                                                        ^
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx: In instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]':
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104:   required from here
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104: error: explicit instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]' but no definition available [-fpermissive]
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx: In instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]':
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104:   required from here
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104: error: explicit instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]' but no definition available [-fpermissive]
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx: In instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]':
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104:   required from here
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104: error: explicit instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]' but no definition available [-fpermissive]
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx: In instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]':
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104:   required from here
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104: error: explicit instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]' but no definition available [-fpermissive]
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx: In instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]':
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104:   required from here
/home/tklemenz/AliSoftware/sw/SOURCES/QualityControl/clusterTask/0/Modules/TPC/src/Clusters.cxx:50:104: error: explicit instantiation of 'void o2::tpc::painter::fillHistogram2D(TH2&, const o2::tpc::CalDet<T>&, o2::tpc::Side) [with T = float]' but no definition available [-fpermissive]
Modules/TPC/CMakeFiles/QcTPC.dir/build.make:147: recipe for target 'Modules/TPC/CMakeFiles/QcTPC.dir/src/Clusters.cxx.o' failed
make[2]: *** [Modules/TPC/CMakeFiles/QcTPC.dir/src/Clusters.cxx.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:2143: recipe for target 'Modules/TPC/CMakeFiles/QcTPC.dir/all' failed
make[1]: *** [Modules/TPC/CMakeFiles/QcTPC.dir/all] Error 2
Makefile:157: recipe for target 'all' failed
make: *** [all] Error 2`

Reply:
I put the line in but get an error, see [here](https://cernbox.cern.ch/index.php/s/6eR2RgDvaU9nNFx).

Reply:
My bad. This cannot work, indeed. Then let's wait for https://github.com/AliceO2Group/AliceO2/pull/4111 and see if this solves the issue.